packages/core/src/lib/pagination/lines.mts

```diff
@@ -0,0 +1,43 @@
+import { rotate, splitLines } from '../utils.mjs';
+import { Paged } from './types.mjs';
```

Comment:
I don't find this name particularly descriptive. I think this is an area around which I'd try to simplify the implementation (the data structure passed around feels large.)

Reply:
Just to be sure - are you concerned that `Paged<T>` is large? I'm a little unsure as it's made of 3 properties, one of them being optional. I understand that the name needs work; I'll put more thought into it and come up with a revision.